sparse/core.py

```diff
@@ -603,55 +611,346 @@ def elemwise_binary(self, func, other, *args, **kwargs):
+            data_list.append(self_unmatched_func)
+            coords_list.append(self_unmatched_coords)
+
+            if self_shape != result_shape:
+                self_broadcast_coords, self_broadcast_func = \
+                    self._get_broadcast_coords_data(self_coords[:, matched_self],
+                                                    matched_coords,
+                                                    self_func[matched_self],
+                                                    self_params,
+                                                    result_shape)
+
+                data_list.append(self_broadcast_func)
+                coords_list.append(self_broadcast_coords)
+
+        other_func = func(0, other_data, *args, **kwargs)
+
+        if (other_func != 0).any():
+            other_unmatched_coords, other_unmatched_func = \
+                self._get_expanded_coords_data(other_coords[:, unmatched_other],
+                                               other_func[unmatched_other],
+                                               other_params,
+                                               result_shape)
+
+            coords_list.append(other_unmatched_coords)
+            data_list.append(other_unmatched_func)
+
+            if other_shape != result_shape:
+                other_broadcast_coords, other_broadcast_func = \
+                    self._get_broadcast_coords_data(other_coords[:, matched_other],
+                                                    matched_coords,
+                                                    other_func[matched_other],
+                                                    other_params,
+                                                    result_shape)
+                data_list.append(other_broadcast_func)
+                coords_list.append(other_broadcast_coords)
 
         # Concatenate matches and mismatches
-        data = np.concatenate([func(self_data[matched_self],
-                                    other_data[matched_other],
-                                    *args, **kwargs),
-                               func(self_data[unmatched_self], 0,
-                                    *args, **kwargs),
-                               func(0, other_data[unmatched_other],
-                                    *args, **kwargs)])
-        coords = np.concatenate([self_coords[matched_self],
-                                 self_coords[unmatched_self],
-                                 other_coords[unmatched_other]])
+        data = np.concatenate(data_list) if len(data_list) else np.empty((0,), dtype=self.dtype)
+        coords = np.concatenate(coords_list, axis=1) if len(coords_list) else \
+            np.empty((0, len(result_shape)), dtype=self.coords.dtype)
 
         nonzero = data != 0
         data = data[nonzero]
-        coords = coords[nonzero]
+        if len(data) != 0:
+            coords, idx = np.unique(coords[:, nonzero], axis=1, return_index=True)
+            data = data[idx]
```

Comment:
Supporting the shapes like `(1, 5)` and `(5, 1)` which have no "common" dimensions is proving to be quite a pain. For some reason, without this check, there were duplicates in `coords` which should not have been summed for these shapes. I still have to figure out why. In the meantime, we have this workaround.

Reply:
As a side comment, the `np.unique call takes up most of the time.`